mycroft/skills/intent_service.py

```diff
@@ -333,12 +361,23 @@ def _converse(self, utterances, lang, message):
                               if time.time() - skill[
                                   1] <= self.converse_timeout * 60]
 
+
+        # first check for system levl skills
+        tmp_active_skills = copy(self.active_skills)
+        for skill in tmp_active_skills:
+            if skill[2] == 'system':
+                if self.do_converse(utterances, skill[0], lang, message):
+                    # update timestamp, or there will be a timeout where
+                    # intent stops conversing whether its being used or not
+                    return IntentMatch('Converse', None, None, skill[0])
+
         # check if any skill wants to handle utterance
-        for skill in copy(self.active_skills):
+        for skill in tmp_active_skills:
```

Comment:
Should this exclude the entries where `skill[2] == system`? seems like those gets called twice now...

Reply:
I thought the above loop would cause us to exit the method on a match, are you saying if line 372 gets executed " return IntentMatch('Converse', None, None, skill[0])" we would still fall thru to the below loop?

Reply:
I could be reading the code wrong but I mean if none of the active_skills with the `system` category has a match (`do_converse()` returns `False` for all of them)

We would continue and run the whole list (including the skills with a `system` category) since they are still in the list, It wouldn't make any practical error since they'd return `False` again, but it'd take some extra time and isn't really needed to be performed again

Reply:
I agree, I was just saying they would not run twice as the above loop would exit if a match occurred however you are correct that we could check if they were system level in the below loop to avoid wasting time checking them again, so yes i agree

Reply:
Also note this PR is related to 

https://github.com/MycroftAI/skill-npr-news/pull/111

and 

https://github.com/MycroftAI/skill-alarm/pull/112